omni/omni.py

```
@@ -0,0 +1,865 @@
+                    await log_channel.send(embed=embed, view=self._create_jump_view(message))
+        except Exception as e:
+            raise RuntimeError(f"Failed to log message: {e}")
+
+    async def periodic_save(self):
+        """Periodically save in-memory statistics to persistent storage."""
+        while True:
+            await asyncio.sleep(self.save_interval)
+            try:
+                await self._save_statistics()
+            except Exception as e:
+                raise RuntimeError(f"Failed to save statistics: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)